Scripts/Mobiles/AI/MageAI.cs

```diff
@@ -790,53 +775,77 @@ public override bool DoActionGuard()
 
         public override bool DoActionFlee()
         {
-            Mobile c = this.m_Mobile.Combatant as Mobile;
+            Mobile c = m_Mobile.Combatant as Mobile;
 
-            if ((this.m_Mobile.Mana > 20 || this.m_Mobile.Mana == this.m_Mobile.ManaMax) && this.m_Mobile.Hits > (this.m_Mobile.HitsMax / 2))
+            if ((m_Mobile.Mana > 20 || m_Mobile.Mana == m_Mobile.ManaMax) && m_Mobile.Hits > (m_Mobile.HitsMax / 2))
             {
-                this.m_Mobile.DebugSay("I am stronger now, my guard is up");
-                this.Action = ActionType.Guard;
+                // If I have a target, go back and fight them
+                if (c != null)
+                {
+                    m_Mobile.DebugSay("I am stronger now, reengaging {0}", c.Name);
+                    Action = ActionType.Combat;
+                }
```

Comment:
Roger that. I've expanded it a lot, but I'll go back and fix up that check on each file with a range check. I just didn't think about it because it's checked at the previous tick while still fleeing.